.github/workflows/publish.yml

```diff
@@ -49,58 +53,69 @@ defaults:
 
 jobs:
     unit-tests:
+        if: |
+            !contains(fromJson(vars.SKIP_UNIT_TESTS), inputs.package) &&
+            inputs.skip-unit-tests == false
         uses: ./.github/workflows/_unit-tests.yml
         with:
             package: ${{ inputs.package }}
             branch: ${{ inputs.branch }}
 
     integration-tests:
         if: |
-            (inputs.package != 'dbt-adapters' && inputs.package != 'dbt-tests-adapter') &&
+            !contains(fromJson(vars.SKIP_INTEGRATION_TESTS), inputs.package) &&
             inputs.skip-integration-tests == false
         uses: ./.github/workflows/_integration-tests.yml
         with:
             package: ${{ inputs.package }}
             branch: ${{ inputs.branch }}
-            repository: ${{ github.repository }}
         secrets: inherit
 
-    generate-changelog:
+    publish-prep-checks:
+        name: "Publish prep checks"
+        if: always()
```

Comment:
Shouldn't we skip this if integration/unit tests fail?

Reply:
This will fail if integration/unit tests fail and will pass if they pass or are skipped. The failure of this job prevents downstream jobs from running because of `if: always() && needs.publish-prep-checks.result == 'success'`. I think this is the behavior we want.